File: common/src/main/java/com/shop/common/TokenUtil.java
package com.shop.common;

import cn.hutool.jwt.JWT;
import cn.hutool.jwt.JWTUtil;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * JWT的封装用于更易用
 * @author lc
 */
public class TokenUtil  {
    /** 密钥串 **/
    public static String SECRET_KEY = "YOUR_SECRET_KEY";
    private static final  String USER_ID_KEY = "USER_ID";
    private static final  String EXPIRE_TIME_KEY = "EXPIRE_TIME";
    /**
     * 获取Token
     */
    public static String getToken(Serializable userId){
        Map<String, Object> map = new HashMap<>();
        map.put(USER_ID_KEY, userId);
        map.put(EXPIRE_TIME_KEY, System.currentTimeMillis() + 1000 * 60 * 60 * 24 * 3);
        return JWTUtil.createToken(map, SECRET_KEY.getBytes());
    }

    /**
     * 获取用户id
     */
    public static String getUserId(String token){
        JWT jwt = JWTUtil.parseToken(token);
        if (!JWTUtil.verify(token, SECRET_KEY.getBytes())) {
            throw new ServiceException("无效Token!");
        }
        Object payload = jwt.getPayload(EXPIRE_TIME_KEY);
        System.out.println(payload);
        if (payload == null || System.currentTimeMillis() > Long.parseLong(payload.toString())) {
            throw new ServiceException("登入失效!");
        }
        return jwt.getPayload(USER_ID_KEY).toString();
    }

    public static void main(String[] args) {
        //测试
        String token = getToken(123456);
        System.out.println(token);
        String userId = getUserId(token);
        System.out.println(userId);
    }
}

File: src/main/kotlin/org/aiwan/util/TokenUtil.kt
package org.aiwan.util

import cn.hutool.core.convert.NumberWithFormat
import cn.hutool.jwt.JWTUtil
import java.io.Serializable

/**
 * JWT的封装用于更易用
 *
 * @author lc
 * @since 7/11/22
 */
object TokenUtil {
    /**
     * 密钥串
     */
    var SECRET_KEY: String = "123456"
    private const val USER_ID_KEY = "USER_ID"
    private const val EXPIRE_TIME_KEY = "EXPIRE_TIME"

    /**
     * 获取Token
     *
     * @since 7/11/22
     */
    fun getToken(userId: Serializable?): String {
        val map: MutableMap<String, Any?> = HashMap<String, Any?>()
        map.put(USER_ID_KEY, userId)
        map.put(EXPIRE_TIME_KEY, System.currentTimeMillis() + 1000L * 60 * 60 * 24 * 365)
        return JWTUtil.createToken(map, SECRET_KEY.toByteArray())
    }

    /**
     * 获取用户id
     *
     * @since 7/11/22
     */
    fun getUserId(token: String): String {
        try {
            if (token.isBlank()) {
                throw MyException("无效Token!", -10)
            }
            val jwt = JWTUtil.parseToken(token)
            if (!JWTUtil.verify(token, SECRET_KEY.toByteArray())) {
                throw MyException("无效Token!", -10)
            }
            val payload = jwt.getPayload(EXPIRE_TIME_KEY)
            if (payload == null || System.currentTimeMillis() > (payload as NumberWithFormat).toLong()) {
                throw MyException("登入失效!", -10)
            }
            return jwt.getPayload(USER_ID_KEY).toString()
        } catch (e: Exception) {
            throw MyException("无效Token!", -10)
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        //测试
        val token = getToken(123456)
        println(token)
        val userId = getUserId(token)
        println(userId)
    }
}
